hydra-main/src/main/java/com/addthis/hydra/minion/HostnameMatcherHostLocationInitializer.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.addthis.hydra.minion;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.addthis.basis.util.Parameter;
+
+import org.codehaus.jackson.annotate.JsonCreator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HostnameMatcherHostLocationInitializer implements HostLocationInitializer{
+    private static final Logger log = LoggerFactory.getLogger(HostnameMatcherHostLocationInitializer.class);
+
+    @JsonCreator
+    public HostnameMatcherHostLocationInitializer() {
+        log.info("using HostnameMatcherHostLocationInitializer.");
+    }
+
+    public HostLocation getHostLocation() {
+
+        String hostname = Parameter.value("minion.localhost");
+
```

Comment:
I'm not comfortable that the regex below are sufficient. Let's discuss how to make it better.

Reply:
To recap our discussion,
- We should by default not have any regex, it should be specified in config
- Users should not be required to provide a regex for each availability property.
- The regexes should be more specific, for example for AD, something like this for OCI: `^[^\.]+\.(ad[\.]+).+`